### PR TITLE
Improves word midpoint selection algorithm

### PR DESCRIPTION
**Pull request by peti123x**
Improves word midpoint selection algorithm to account for other languages like German, which sees unusually long words where we do not necessarily want to actually select half of the word. 

The chosen algorithm is as advised is ceil(log2(word.length)) which scales much better with long words but will, for the most part, mimic the same midpoints-ish as previously. 

In following PRs we can create additional 'fixation' algorithms, e.g. we would ceil or floor depending on low or strong fixation, or a completely different approach could be taken where words are categorised based on how long they are (short, medium, long) and individual cutoff point algorithms are chosen per bucket.

### CODE REVIEW

**Review by Classic-Paterson — APPROVED** [review]

**Comment by X140Yu** [comment]
Hi @peti123x 

I think this PR should be better done with the "change fixation" implemented, or it will break the en-US words and make the bolded part too short

File: src/convert.js [hunk 1]
```diff
@@ -11,13 +19,13 @@ function highlightText(sentenceText) {
       if (hasNumber.test(word)) {
         return word;
       }
-      const { length } = word;
-      let midPoint = 1;
-      if (length > 3) midPoint = Math.round(length / 2);
+
+      const midPoint = determineMidpoint(word);
```
Comment (X140Yu) [inline src/convert.js:23]: can skip with `word.length <= 3`, we only want to bold the first letter of these words

**Comment by peti123x** [comment]
Hi @X140Yu 

Thanks for the feedback, will take a look at implementing shortly.

File: src/convert.js (hunk 1 above)
Comment (peti123x) [reply src/convert.js:23]: I'm not convinced this should be hardcoded, if change fixation is implemented then we should let the algorithm decide, no?
The difference is pretty much `Math.ceil(x)` vs. `Math.floor()` which will produce the difference of 1 character that we see here on 3 letter words. Strong fixation would be 2 letters bolded and weak is 1 letters.

Comment (Alevale) [reply src/convert.js:23]: > Strong fixation would be 2 letters bolded and weak is 1 letters.

Maybe that is something that could be delegated to a setting of the chrome extension?

**Comment by asieduernest12** [comment]
Thank you @peti123x i believe we can close this request, but i see your point and the point of @Alevale too.
Fixations is now implemented and you could probably make a new pull request to account for exceeding long words in other languages.

If you do kindly make something like function 
`
/**@returns number*/
function getMidPoint(/**number*/){
  // body goes here
}`

**Pull request closed without merging**